File: leetcode-04/solutions/cpp/0124.cpp
class Solution {
 public:
  int maxPathSum(TreeNode* root) {
    helper(root);

    return ans;
  }

 private:
  int ans = INT_MIN;

  int helper(TreeNode* root) {
    if (!root) return 0;

    int left = max(helper(root->left), 0);
    int right = max(helper(root->right), 0);
    ans = max(ans, root->val + left + right);

    return root->val + max(left, right);
  }
};

File: leetcode-04/solutions/python3/0124.py
class Solution:
    def maxPathSum(self, root: TreeNode) -> int:
        self.helper(root)

        return self.ans

    ans = float('-inf')

    def helper(self, root: TreeNode) -> int:
        if not root:
            return 0

        left = max(self.helper(root.left), 0)
        right = max(self.helper(root.right), 0)
        self.ans = max(self.ans, root.val + left + right)

        return root.val + max(left, right)
